lib/elixir/test/elixir/exception_test.exs

```diff
@@ -482,6 +482,42 @@ defmodule ExceptionTest do
                  "such as map.field or module.function, make sure the left side of the dot is an atom or a map"
     end
 
+    test "annotates key error with suggestions" do
+      message = blame_message(%{first: nil, second: nil}, fn map -> map.firts end)
+
+      assert message == """
+             key :firts not found in: %{first: nil, second: nil}. Did you mean one of:
+
+                   * :first
+             """
+
+      message = blame_message(%{"first" => nil, "second" => nil}, fn map -> map.firts end)
```

Comment:
We should not compare atoms with strings. After all, `%{"first" => nil}.first` also wouldn't work. I think for now we should only provide did you mean on atom access. We can always expand later.